src/configure/clients/github/githubClient.ts

```diff
@@ -1,69 +1,150 @@
 import { UrlBasedRequestPrepareOptions } from 'ms-rest';
+import { stringCompareFunction } from "../../helper/commonHelper";
 import { GitHubProvider } from "../../helper/gitHubHelper";
 import { SodiumLibHelper } from "../../helper/sodium/SodiumLibHelper";
+import { GitHubOrganization, GitHubRepo } from '../../model/models';
 import { RestClient } from "../restClient";
+const uuid = require('uuid/v4');
 
 const UserAgent = "deploy-to-azure-vscode";
 
 export class GithubClient {
 
     private patToken: string;
     private url: string;
+    private listOrgPromise: Promise<void | GitHubOrganization[]>;
+
 
     constructor(patToken: string, remoteUrl: string) {
         this.patToken = patToken;
         this.url = remoteUrl;
+        this.listOrgPromise = this.listOrganizations(this.patToken);
     }
 
-    public async createOrUpdateGithubSecret(secretName: string, body: string): Promise<void> {
-        let secretKeyObject: GitHubSecretKey = await this._getGitHubSecretKey();
-        let sodiumObj = new SodiumLibHelper(secretKeyObject.key);
-        let encryptedBytes: Uint8Array = sodiumObj.encrypt(body);
-        let encryptedBytesAsString: string = SodiumLibHelper.convertUint8ArrayToString(encryptedBytes);
-        let encryptedEncodedText = SodiumLibHelper.encodeToBase64(encryptedBytesAsString);
-        await this._setGithubSecret(secretName, secretKeyObject.key_id, encryptedEncodedText);
+    public createGithubRepo(orgName: string, repoName: string): Promise<void | GitHubRepo> {
```

Comment:
why `void | GitHubRepo` and not just `GitHubRepo`?

Reply:
Since it's an API call, due to some error (like existing repo name) the function might not return a GitHubRepo object. Therefore, the return type 'void' was needed to be added. Also, by using 'void', it becomes convenient to move to other cases, where organization name or uuid needs to be appended in order to give a unique name to the GitHub repository.